tap_klaviyo_custom/streams.py

```diff
@@ -1,157 +1,122 @@
-    def prepare_request(
-        self, context: Optional[dict], next_page_token: Optional[Any], list_id: Optional[str]
-    ) -> requests.PreparedRequest:
-        """Prepare a request object.
-
-        If partitioning is supported, the `context` object will contain the partition
-        definitions. Pagination information can be parsed from `next_page_token` if
-        `next_page_token` is not None.
-        """
-        http_method = self.rest_method
-        url: str = self.get_url(context, list_id)
-        params: dict = self.get_url_params(context)
-        request_data = self.prepare_request_payload(context, next_page_token)
-        headers = self.http_headers
-
-        if next_page_token != None:
-            params['marker'] = next_page_token
-
-        request = cast(
-            requests.PreparedRequest,
-            self.requests_session.prepare_request(
-                requests.Request(
-                    method=http_method,
-                    url=url,
-                    params=params,
-                    headers=headers,
-                    json=request_data,
-                )
-            ),
-        )
-        # sleep timer to avoid Klaviyo API rate limit errors
-        time.sleep(1)
-        return request
+    def get_next_page_token(
+        self, response: requests.Response, previous_token: Optional[Any]
+    ) -> Any:
+        """Return token identifying next page or None if all records have been read."""
+        # since we're using partitions, there's only ever one page per partition
+        return None
 
-    def get_url(self, context: Optional[dict], list_id: Optional[str]) -> str:
-        """Return a URL, optionally targeted to a specific partition or context.
+    def post_process(self, row: dict, context: Optional[dict] = None) -> Optional[dict]:
+        """As needed, append or transform raw data to match expected structure."""
+        # For some reason,  the id is not included in the response
+        assert context is not None
+        return {**row, "list_id": context["list_id"]}
 
-        Developers override this method to perform dynamic URL generation.
-        """
-        url = self.url_base+self.path.format(list_id=list_id)
+    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
+        """Return a context dictionary for child streams."""
+        assert context is not None
+        return {"list_id": context["list_id"]}
 
-        return url
 
+class ListMembersStream(KlaviyoStream):
+    """A Stream for list members."""
 
-    def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
-        """Return a generator of row-type dictionary objects.
+    name = "list_members"
+    path = "group/{list_id}/members/all"
+    primary_keys = ["id"]
+    replication_key = None
+    schema_filepath = SCHEMAS_DIR / "list_members.json"
+    records_jsonpath = "$.records[*]"
+    next_page_token_jsonpath = "$.marker"
+    parent_stream_type = ListsStream
```

Comment:
Oh wait, I see here that ListsStream does get listed as a parent_stream. What's the point of doing it here rather than in the class declaration?

Reply:
Where would it go in the class declaration? AFAICT [from the docs](https://gitlab.com/meltano/sdk/-/blob/main/docs/parent_streams.md) this is the way to do it

Reply:
That's on me, I think I better understand how parent-child streams work now. My brain jumped to assume they work the same as regular class inheritance, where you include the parent class in the child class declaration. But I see now how that is more for passing along behavior and methods, whereas it seems like parent-child streams are specifically for passing along the context. Is that correct?

Reply:
With that being said, this all looks good to me and ready to merge

Reply:
Yeah inheritance can be used to share common functionality between streams, but it wouldn't work for parent-child streams. I see how that could be confusing though. I'll update [the documentation issue](https://gitlab.com/meltano/sdk/-/issues/298) I already opened so maybe they can make those names clearer